fe/fe-core/src/main/java/org/apache/doris/service/FrontendServiceImpl.java

```diff
@@ -2296,15 +2297,16 @@ public TFrontendPingFrontendResult ping(TFrontendPingFrontendRequest request) th
 
     @Override
     public TFetchSchemaTableDataResult fetchSchemaTableData(TFetchSchemaTableDataRequest request) throws TException {
-        switch (request.getSchemaTableName()) {
-            case METADATA_TABLE:
-                return MetadataGenerator.getMetadataTable(request);
-            case SCHEMA_TABLE:
-                return MetadataGenerator.getSchemaTableData(request);
-            default:
-                break;
+        if (!request.isSetSchemaTableName()) {
+            return MetadataGenerator.errorResult("Fetch schema table name is not set");
+        }
+        // tvf queries
+        if (request.getSchemaTableName() == TSchemaTableName.METADATA_TABLE) {
+            return MetadataGenerator.getMetadataTable(request);
+        } else {
```

Comment:
change to  else if (request.getSchemaTableName() == TSchemaTableName.INFORMATION_SCHEMA) {
            
        } else {
                throw new TException("xxxxx")
        }

Then we could make sure if we add another ENUM, the logic will not be wrong.